ansible_collections/arista/avd/roles/eos_designs_to_containerlab/tasks/deploy.yml

```diff
@@ -0,0 +1,237 @@
+---
+#tasks file for containerlab hosts
+
+- ansible.builtin.setup:
+    # gather_subset:
+    #   - '!all'
+    filter:
+      - 'ansible_env'
+
+- name: Check if more than one containerlab host is defined
+  set_fact:
+    multinode: "{{ true if groups['CONTAINERLAB']|length > 1 else false }}"
+  run_once: true
+  delegate_to: localhost
+
+- name: Check if there is already a topology file with the same name
+  stat:
+    path: "{{ ansible_facts.env.PWD }}/{{ containerlab_labname }}/{{inventory_hostname}}_{{ containerlab_configuration }}"
+  register: stat_result
+
+- name: Destroy the lab using containerlab in a container
+  become: yes
+  command: "docker run \
+    --rm \
+    -it \
+    --privileged \
+    --network host \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -v /var/run/netns:/var/run/netns \
+    -v /etc/hosts:/etc/hosts \
+    --pid='host' \
+    -v {{ ansible_facts.env.PWD }}:{{ ansible_facts.env.PWD }} \
+    -w {{ ansible_facts.env.PWD }}/{{ containerlab_labname }} \
+    ghcr.io/srl-labs/clab clab destroy -t {{inventory_hostname}}_{{ containerlab_configuration }} --cleanup "
+  when: containerlab_mode == 'container' and stat_result.stat.exists
+
+- name: Destroy the lab using installed containerlab 
+  become: yes
+  command: "clab destroy -t {{inventory_hostname}}_{{ containerlab_configuration }} --cleanup"
+  args:
+    chdir: "{{ ansible_facts.env.PWD }}/{{ containerlab_labname }}"
+  when: containerlab_mode == 'installed' and stat_result.stat.exists
+
+- name: Include connections per containerlab host variables
+  delegate_to: localhost
+  include_vars: 
+    file: "{{ containerlab_dir }}/containerlab_host_connections.yml"
+    name: connections_per_host
+  run_once: true
+
+- name: Remove the temporary file with connections per containerlab host
+  delegate_to: localhost
+  file:
+    path: "{{ containerlab_dir }}/containerlab_host_connections.yml"
+    state: absent
+  run_once: true
+
+- name: Include AVD nodes
+  delegate_to: localhost
+  include_vars: 
+    file: "{{ containerlab_dir }}/temp_AVD_ansible_hosts.yml"
+    name: avd_nodes_all
+  run_once: true
+
+- name: Remove the temporary file with AVD hosts
+  delegate_to: localhost
+  file:
+    path: "{{ containerlab_dir }}/temp_AVD_ansible_hosts.yml"
+    state: absent
+  run_once: true
+
+- name: Generate temporary containerlab host to avd nodes mapping
```

Comment:
these three tasks could be replaced with `yaml_templates_to_facts` where we just set facts directly from a template